Index: src/osx/carbon/graphics.cpp
===================================================================
--- src/osx/carbon/graphics.cpp	(revision 63701)
+++ src/osx/carbon/graphics.cpp	(working copy)
@@ -1380,14 +1380,7 @@
 
     virtual bool ShouldOffset() const
     {
-        int penwidth = 0 ;
-        if ( !m_pen.IsNull() )
-        {
-            penwidth = (int)((wxMacCoreGraphicsPenData*)m_pen.GetRefData())->GetWidth();
-            if ( penwidth == 0 )
-                penwidth = 1;
-        }
-        return ( penwidth % 2 ) == 1;
+        return false;
     }
     //
     // text
