Index: src/osx/carbon/graphics.cpp
===================================================================
--- src/osx/carbon/graphics.cpp	(revision 63590)
+++ src/osx/carbon/graphics.cpp	(revision 63589)
@@ -1461,18 +1461,14 @@
         m_cg = cg;
         m_offset = offset;
         if ( m_offset )
-        {
-            m_userOffset = CGContextConvertSizeToUserSpace( m_cg, CGSizeMake( 0.5 , 0.5 ) );
-            CGContextTranslateCTM( m_cg, m_userOffset.width , m_userOffset.height );
-        }
+            CGContextTranslateCTM( m_cg, (CGFloat) 0.5, (CGFloat) 0.5 );
     }
     ~wxQuartzOffsetHelper( )
     {
         if ( m_offset )
-            CGContextTranslateCTM( m_cg, -m_userOffset.width , -m_userOffset.height );
+            CGContextTranslateCTM( m_cg, (CGFloat) -0.5, (CGFloat) -0.5 );
     }
 public :
-    CGSize m_userOffset;
     CGContextRef m_cg;
     bool m_offset;
 } ;
