Index: interface/wx/defs.h
===================================================================
--- interface/wx/defs.h	(revision 69933)
+++ interface/wx/defs.h	(working copy)
@@ -886,7 +886,19 @@
     WXK_CLEAR,
     WXK_SHIFT,
     WXK_ALT,
+    /** Note that under Mac OS X, to improve compatibility with other
+      * systems, 'WXK_CONTROL' represents the 'Command' key. Use this
+      * constant to work with keyboard shortcuts. See 'WXK_RAW_CONTROL'
+      * to get the state of the actual 'Control' key.
+      */
     WXK_CONTROL,
+    /** Under Mac OS X, where the 'Command' key is mapped to 'Control'
+      * to improve compatibility with other systems, WXK_RAW_CONTROL may
+      * be used to obtain the state of the actual 'Control' key
+      * ('WXK_CONTROL' would obtain the status of the 'Command' key).
+      * Under Windows/Linux/Others, this is equivalent to WXK_RAW_CONTROL
+      */
+    WXK_RAW_CONTROL,
     WXK_MENU,
     WXK_PAUSE,
     WXK_CAPITAL,
@@ -977,6 +989,11 @@
     WXK_WINDOWS_LEFT,
     WXK_WINDOWS_RIGHT,
     WXK_WINDOWS_MENU ,
+    
+    /** This special key code represents the key used for keyboard shortcuts. Under Mac OS X,
+      * this key maps to the 'Command' (aka logo or 'Apple') key, whereas on Linux/Windows/others
+      * this is the Control key
+      */
     WXK_COMMAND,
 
     /** Hardware-specific buttons */
Index: src/osx/core/hid.cpp
===================================================================
--- src/osx/core/hid.cpp	(revision 69933)
+++ src/osx/core/hid.cpp	(working copy)
@@ -601,7 +601,7 @@
 
             //Menu keys, Shift, other specials
             case kHIDUsage_KeyboardLeftControl:
-                AddCookie(CFArrayGetValueAtIndex(Array, i),WXK_CONTROL);
+                AddCookie(CFArrayGetValueAtIndex(Array, i),WXK_RAW_CONTROL);
                 break;
             case kHIDUsage_KeyboardLeftShift:
                 AddCookie(CFArrayGetValueAtIndex(Array, i),WXK_SHIFT);
@@ -610,7 +610,7 @@
                 AddCookie(CFArrayGetValueAtIndex(Array, i),WXK_ALT);
                 break;
             case kHIDUsage_KeyboardLeftGUI:
-                AddCookie(CFArrayGetValueAtIndex(Array, i),WXK_MENU);
+                AddCookie(CFArrayGetValueAtIndex(Array, i),WXK_CONTROL);
                 break;
             case kHIDUsage_KeyboardRightControl:
                 AddCookie(CFArrayGetValueAtIndex(Array, i),WXK_RCONTROL);
