Index: src/osx/carbon/toolbar.cpp
===================================================================
--- src/osx/carbon/toolbar.cpp	(revision 63701)
+++ src/osx/carbon/toolbar.cpp	(working copy)
@@ -1322,6 +1322,9 @@
                             wxFAIL_MSG("Reference count of native tool was illegal before removal");
                         }
                         wxASSERT( IsValidControlHandle(tool->GetControl()->GetPeer()->GetControlRef() )) ;
+                        
+                        wxString label = tool->GetLabel();
+                        HIToolbarItemSetLabel( hiItemRef, wxCFStringRef(label, GetFont().GetEncoding()) );
                     }
                 }

 
